pandas/_libs/lib.pyx

```diff
@@ -1291,44 +1291,49 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
 
     if util.is_array(value):
         values = value
-    elif hasattr(value, 'dtype'):
+    elif hasattr(value, "inferred_type") and skipna is False:
+        # Index, use the cached attribute if possible, populate the cache otherwise
+        return value.inferred_type
+    elif hasattr(value, "dtype"):
         # this will handle ndarray-like
         # e.g. categoricals
-        try:
-            values = getattr(value, '_values', getattr(value, 'values', value))
-        except TypeError:
-            # This gets hit if we have an EA, since cython expects `values`
-            #  to be an ndarray
-            value = _try_infer_map(value)
+        dtype = value.dtype
+        if not isinstance(dtype, np.dtype):
+            value = _try_infer_map(value.dtype)
             if value is not None:
                 return value
 
-            # its ndarray like but we can't handle
+            # its ndarray-like but we can't handle
             raise ValueError(f"cannot infer type for {type(value)}")
 
+        # Unwrap Series/Index
+        values = np.asarray(value)
+
     else:
         if not isinstance(value, list):
             value = list(value)
-        from pandas.core.dtypes.cast import (
-            construct_1d_object_array_from_listlike)
-        values = construct_1d_object_array_from_listlike(value)
+
+        # See also: construct_1d_object_array_from_listlike
+        values = np.empty(len(value), dtype=object)
```

Comment:
why did you change from using the function for this?

Reply:
trying to untangle dependency structure

Reply:
can revert, this isnt related to the rest of the PR

Reply:
that function exists for a reason, to avoid constantly repeating the 2 lines you added. pls revert.

Reply:
updated + green